src/notifications/notificationmanager.cpp

```diff
@@ -70,28 +77,92 @@ const int DefaultNotificationPriority = 50;
 const int CommitDelay = 10 * 1000;
 const int PublicationDelay = 1000;
 
-QString getProcessName(uint pid)
+static bool processIsPrivileged(int pid)
+{
+    bool isPrivileged = false;
+    if (pid == getpid()) {
+        // Internal operations are considered privileged
+        isPrivileged = true;
+    } else if (pid > 0) {
+        QFileInfo info(QString("/proc/%1").arg(pid));
+        isPrivileged = info.group() == QLatin1String("privileged") || info.owner() == QLatin1String("root");
+    }
+    NOTIFICATIONS_DEBUG("pid" << pid << "-> isPrivileged" << isPrivileged);
+    return isPrivileged;
+}
+
+static QString getProcessCmdline(int pid)
+{
+    QString cmdline;
+    if (pid > 0) {
+        const QString path(QString::fromLatin1("/proc/%1/cmdline").arg(QString::number(pid)));
+        QFile file(path);
+        if (file.open(QIODevice::ReadOnly)) {
+            const QByteArray data = file.readAll();
+            cmdline = QFile::decodeName(data.left(data.indexOf('\0')));
```

Comment:
Decoding command line, or similar, data with filesystem encoding feels a bit random. Maybe just QString::fromUtf8()?

Reply:
This is most likely dealing with path-like string in not strongly defined encoding + original code used that decode, so I kept it without putting too much thought into it. Probably decode ends up using utf8 anyway, and explicit behavior leaves less room for guessing -> will change.